src/standalone/Soil/energy_hydrology.jl

```diff
@@ -2,54 +2,53 @@
     EnergyHydrologyParameters{FT <: AbstractFloat}
 
 A parameter structure for the integrated soil water and energy
- equation system. In this simplest
-form, we assume the conductivity and volumetric heat capacity
-of the soil are constant.
+ equation system. 
 $(DocStringExtensions.FIELDS)
 """
 struct EnergyHydrologyParameters{
     FT <: AbstractFloat,
-    C <: AbstractSoilHydrologyClosure,
+    F <: Union{AbstractFloat, ClimaCore.Fields.Field},
```

Comment:
This doesn't guarantee that the struct is concretely typed. F could be `AbstractFloat`, which is not concrete. What you want is `F <: Union{<: Real, <: ClimaCore.Fields.Field}` I think.
Or
`F <: Union{FT, FIELD} where {FIELD <: ClimaCore.Fields.Field}`

Reply:
Based off of this: https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type, this seems correct now.

I went with `F <: Union{<: AbstractFloat, <: ClimaCore.Fields.Field}`, because `Real` is an abstract type that includes ints as concrete examples, which isnt what we want.

Reply:
good catch!

Reply:
In Julia, `Union` and `Tuple` are special in terms of how their type parameters are treated. Here are some examples
```
julia> 1 isa Union{Real, Fields.Field}
true
julia> (1,) isa Tuple{Real}
true
julia> [1] isa Vector{<:Real}
true
julia> [1] isa Vector{Real}
false
```